File: src/main/java/kr/co/bookand/backend/account/repository/AccountRepository.java
package kr.co.bookand.backend.account.repository;

import kr.co.bookand.backend.account.domain.Account;
import kr.co.bookand.backend.common.DeviceType;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface AccountRepository extends JpaRepository<Account, Long> {

    Optional<Account> findByEmail(String email);

    Optional<Account> findByNickname(String nickname);

    boolean existsByEmail(String email);

    boolean existsByNickname(String nickname);

    Long countBy();

    Long countByDeviceType(DeviceType userType);
}

File: src/main/kotlin/kr/co/bookand/backend/account/repository/AccountRepository.kt
package kr.co.bookand.backend.account.repository

import kr.co.bookand.backend.account.model.Account
import org.springframework.data.jpa.repository.JpaRepository

interface AccountRepository : JpaRepository<Account, Long> {
    fun findByEmail(email: String): Account?
    fun findByNickname(nickname: String): Account?
    fun existsByNickname(nickname: String): Boolean
    fun existsByEmail(email: String): Boolean
    fun countAllByVisibility(visibility: Boolean): Long
}